### ISSUE TITLE
ms writer: Unicode char hex figures should be uppercase (or translation dropped)

### ISSUE THREAD

**Issue by teoric** [jgm/pandoc#4597]
Pandoc (2.1.3) renders some Unicode characters als `groff` characters. Hex figures `A–F` seem to be lower case, and should be upper case (e.g. `¬` is rendered as `\u[00ac]` instead of `\u[00AC]`).

This has a more general scope, if you want: Pandoc does a lot of character translations for the `ms` writer.  Given that it still lets a lot of UTF-8 pass, one must use `groff -Kutf-8` or `groff -Dutf-8` anyway (or, with older groff, using `preconv` directly), and then translations like `–` to `\(en` etc. are mostly handled by groff's `preconv(1)`.  Wouldn't it be better in the long run to minimize character translations?

# Example
The following input is rendered by Pandoc replacing the negation symbol by a unicode-character:
```markdown
This is $\\neg p$
```
i.e.
```sh
echo '$\\neg p$' | pandoc -f markdown  -t ms
```
gives
```nroff
.LP
This is @\[u00ac] p@
```
This, run through `groff -Dutf-8` gives an error message,  `warning: can't find special character 'u00ac'`. Replacing `\[u00ac]` with the simple Unicode `¬` works fine, and so does `\[u00AC]`.

**Comment by jgm** [comment]
This is coming from texmath's eqn writer.  Unicode characters in body text aren't escaped.
I'm reluctant to change texmath's eqn writer, which can be used in principle for man pages and other places where you might not want UTF-8.  But I can fix the upper/lowercase issue.

**Comment by jgm** [comment]
Perhaps an even better solution would be:  have the texmath eqn writer emit UTF-8, and expect the calling program to escape it if that's what is needed.  The transformation is fairly straightforward and reliable.

We should also make the `ms` writer respect `--ascii`, if it doesn't already.

**Comment by teoric** [comment]
Thank you for looking into this!  Well, if the escaping works, any solution is fine with me, of course.  I would assume that doing escaping as late as possible is indeed a good thing nowadays.

(I took this to be the same as en-dashes or elipses, which are escaped even in body text. Try `echo "–…" |pandoc -t ms`, and with no `--ascii`.  Ellipsis should be preserved or changed to `.\|.\|.\&` or something, if possible, because the three dots render differently.)

**Comment by jgm** [comment]
Yes, I'm not sure about passing through em and en dash.
One thing that can be said against it is that these
characters are indistinguishable in most monospace fonts.

What is wrong with `\&...` for ellipses?  It looks better to me
than the more spaced-out version you provide.

**Comment by jgm** [comment]
I'll close this since the basic issue has been addressed.

**Comment by teoric** [comment]
Regarding –/—: I agree, maybe i'm desensitized as I found a good font, and look out for these in the PDFs anyway.  But note that we are talking about people who entered en/em dashes in their editor (i.e., they probably know what to do with them), and then get `\(en` in the ms source.

Regarding ellipsis: In all proportional fonts I know, "…" (ellipsis) is more spaced out than "..." (three dots).  So maybe you prefer the three dots, but the form I suggested looks more like what is usual in print. (As I said, just pass it through, and the font will decide.)
